### PR TITLE
Added Haar random function

### PR DESCRIPTION
**Pull request by babbush**
I find myself writing these lines a lot in some demos I'm making so might as well make it a function.

### CODE REVIEW

File: src/openfermion/utils/_testing_utils.py [hunk 1]
```diff
@@ -23,6 +23,16 @@
                              QuadraticHamiltonian)
 
 
+def haar_random_vector(n, seed=None):
+    """Generate an n dimensional Haar randomd vector."""
+    if seed is not None:
+        numpy.random.seed(seed)
+    vector = numpy.random.randn(n).astype(numpy.complex64)
```
Comment (kevinsung) [inline src/openfermion/utils/_testing_utils.py]: Why are you using single-precision? We don't do that anywhere else in the library.

File: src/openfermion/utils/_testing_utils.py [hunk 2]
```diff
@@ -23,6 +23,16 @@
...
+    vector += 1.j * numpy.random.randn(n).astype(numpy.complex64)
+    normalization = numpy.sqrt(vector.dot(numpy.conjugate(vector)))
```
Comment (kevinsung) [inline src/openfermion/utils/_testing_utils.py]: `numpy.linalg.norm(vector)`

**Comment by babbush** [comment]
I don't know. I switched to double precision. However, I don't want to use that norm function because its not very explicit and I'd prefer to use something that I can easily tell what it is doing. "norm" is ambiguous and can mean many things. So I think my way is more readable.

**Comment by kevinsung** [comment]
The word "norm" is ambiguous but `numpy.linalg.norm` is unambiguous. Anyway, I'll approve.

**Comment by babbush** [comment]
It is totally ambiguous. I'd need to look up numpy.linalg.norm to know what it does!

**Pull request merged**